src/sensai/vector_model.py

```diff
@@ -133,20 +127,113 @@ def withName(self, name: str):
         self.setName(name)
         return self
 
+    def getInputTransformer(self, cls: Type[DataFrameTransformer]):
+        for it in self._inputTransformerChain.dataFrameTransformers:
+            if isinstance(it, cls):
+                return it
+        return None
+
+    def setName(self, name):
+        self._name = name
+
+    def getName(self):
+        if self._name is None:
+            return "unnamed-%s-%x" % (self.__class__.__name__, id(self))
+        return self._name
+
+    def setFeatureGenerator(self, featureGenerator: Optional[FeatureGenerator]):
+        self.withFeatureGenerator(featureGenerator)
+
+    def getFeatureGenerator(self) -> Optional[FeatureGenerator]:
+        return self._featureGenerator
+
+
+class FittableModel(PredictorModel, ABC):
     @abstractmethod
-    def isRegressionModel(self) -> bool:
+    def fit(self, X: pd.DataFrame, Y: pd.DataFrame = None):
+        pass
+
+
+class RulesBasedModel(FittableModel, ABC):
```

Comment:
I don't really see the need for this class, especially since it duplicates functionality from VectorModel for no good reason - and loses some functionality along the way. For instance, the checks done in _computeInputs are omitted here. 
Why wouldn't it be OK to implement a subclass of VectorModel with an empty _fit? After all, a rule-based model _is_ a regular model which just doesn't do fitting. Either way, it would be more natural for this to be a specialisation rather than a sibling in the hierarchy.

Reply:
Implementing empty _fit compromises efficiency as during fit we would call fitGenerate on the feature generators instead if simply fit. I don't see how I can make it a specialization of VectorModel because of that. The RuleBasedModel cannot have the input checks for the above reason

Reply:
I see. If efficiency is more important (which is certainly can be), I guess it's the right decision you made.

Reply:
Then again, don't we usually have time to spare when training?

Reply:
In my use case it is. For the RuleBasedModel the rules are very simple (a sum) but feature generator is costly (processing images). In fact, most feature generation will be rule based itself so fit is for free but fitGenerate is expensive